### ISSUE TITLE
Debugging a semi-functional firmware

### ISSUE THREAD

**Issue by chrisgervang** [stacksmashing/gb-link-firmware#5]
Hey there! 
### TL;DR
I've got a local dev environment set up, but the device is failing to connect to the web client. My goal is to log the bytes sent by my gameboy for a project.

### Symptoms
- Pico LED isn't flashing when using gb-link-firmware (but pico-examples/blink works)
- WebUSB Landing Page Notification fails to pop up
- `chrome://usb-internals/` doesn't list a landing page URL
- web client fails to finish connecting (sends `0x22` but doesn't hear reply)

### Environment
- pico-sdk: latest [master bfcbefa](https://github.com/raspberrypi/pico-sdk/tree/bfcbefafc5d2a210551a4d9d80b4303d4ae0adf7) (version 1.2.0)
- tinyusbb: latest [master 21bfd11](https://github.com/hathach/tinyusb/tree/21bfd11683c4bb6e0996130bd1061a97d3049121)
- os: mac 10.15.4
- browser: chrome 91

### What I've Done So Far

- The [firmware](http://stacksmashing.net/firmware-0.1.uf2) from the website works as expected.
- I got my pico SDK all set, confirmed blink example works. 
- `gb-link-firmware` compiled and flashed onto my Pico successfully. 
- `gb-tetris-web` is up and running locally
- I modified the firmware to land on `localhost:3000` and `http`.
- In chrome, I see the device on `chrome://usb-internals/`:
<img width="372" alt="Screen Shot 2021-07-18 at 5 33 54 PM" src="https://user-images.githubusercontent.com/2461547/126087281-53f9ec38-1c5a-44aa-8aba-b73304a753ad.png">

- On the frontend the console gets to:
<img width="209" alt="Screen Shot 2021-07-18 at 5 36 07 PM" src="https://user-images.githubusercontent.com/2461547/126087923-baa5fa65-4eab-4699-aedb-0441c8850a7f.png">

- I've also tried using the `uf2` made by #1 on their [releases page](https://github.com/hinzundcode/gb-link-firmware/releases). Same issue it seems.

### Research
- This [Adafruit thread](https://github.com/webusb/arduino/issues/69#issuecomment-528097726) shows a `WebUSB Landing Page` property in the usb debugger. Should the pico firmware have this?

**Comment by chrisgervang** [comment]
I think I've worked around this issue!

```
cd pico-sdk
git checkout 1.0.1
git submodule update
```

Then rebuilding everything from scratch worked.

I saw in #4 a mention that the original (working) firmware used pico-sdk 1.0.1, so switching to that and updating tinyusb to the version used at the time does the trick. Expect to see this:

```
git submodule       
> e0aa405d19e35dbf58cf502b8106455c1a3c2a5c lib/tinyusb (0.7.0-9-ge0aa405d)
```

And as expected, `chrome://usb-internals/` now lists my landing page and the notification works.
<img width="421" alt="Screen Shot 2021-07-18 at 10 15 52 PM" src="https://user-images.githubusercontent.com/2461547/126106705-bc4d2185-9b55-40a4-aee4-7578c14ab137.png">
